CDS/js/Datastore.js

```diff
@@ -114,24 +116,100 @@ function get(conn,type, key, callback, userdata) { /* callback as function(err,
                         console.error(err);
                         reject(err);
                     }
+                    var rtn;
                     if (!row) {
-                        fulfill({value: "(value not found)",type: type,key: key});
+                        rtn = {value: "(value not found)",type: type,key: key};
                     } else {
-                        if(callback){
-                            fulfill(callback(row.value,type,key,userdata));
-                        } else {
-                            fulfill({value: row.value,type: type,key: key});
-                        }
+                        rtn = {value: row.value,type: type,key: key}
+                    }
+
+                    if(callback){
```

Comment:
passing in a callback to a promise is a very odd thing to do. Why not use the `.then` method?